src/xen/xen.c

```diff
@@ -780,14 +819,9 @@ status xenstore_sync_printf(u32 tx_id, buffer path, const char *node, const char
 
 status xenstore_read_u64(u32 tx_id, buffer path, const char *node, u64 *result)
 {
-    buffer request = allocate_buffer(xen_info.h, 64);
-    assert(push_buffer(request, path));
-    push_u8(request, '/');
-    assert(buffer_write(request, node, runtime_strlen(node)));
-    push_u8(request, 0);
-
+    *result = 0;
```

Comment:
We can remove this (the result variable should not be touched in case of an error)

Reply:
This line is there to eliminate a compiler uninitialized warning in the callers.

Reply:
OK, I see the uninitialized warning in the callers. This only happens for callers in xen.c, because apparently gcc inlines these calls but somehow doesn't detect that an uninitialized use cannot happen. But adding this line breaks the logic in xenblk_watch_service(), where the sector_size variable is used to read first "sector-size" and then "physical-sector-size": if the second read results in an error, the sector_size variable is now reset to 0, which causes xbd->capacity to be set to 0. So we should either put the initialization in the callers inside xen.c, or make sure that all callers don't rely on the result variable not being touched in case of an error. The same goes for xenstore_transaction_start().

Reply:
Thanks, I missed that usage before. I took out the initialization in those functions and initialized at the caller instead, but at least it seems to only be needed in xen.c.